src/components/stepper/stepper/stepper.ts

```diff
@@ -0,0 +1,277 @@
+
+    return true;
+  }
+
+  private _select(step: SbbStepElement): void {
+    if (!this._isValidStep(step)) {
+      return;
+    }
+    const validatePayload = {
+      currentIndex: this.selectedIndex,
+      currentStep: this.selected,
+      nextIndex: this.selectedIndex !== undefined ? this.selectedIndex + 1 : undefined,
+      nextStep: this.selectedIndex !== undefined ? this.steps[this.selectedIndex + 1] : undefined,
+    };
+    if (this.selected && !this.selected.validate(validatePayload)) {
+      return;
+    }
+    const current = this.selected;
+    if (current && current.label) {
+      current.toggleAttribute('data-selected', false);
+      current.label.internals.ariaSelected = 'false';
+      current.label.toggleAttribute('data-selected', false);
+    }
+    if (step && step.label) {
+      step.toggleAttribute('data-selected', true);
+      step.label.internals.ariaSelected = 'true';
+      step.label.toggleAttribute('data-selected', true);
+      this._setMarkerSize();
+
+      if (this._loaded) {
+        step.label.focus();
+      }
+    }
+  }
+
+  private _setMarkerSize(): void {
+    if (
+      !this._loaded ||
+      !this.selected ||
+      this.selectedIndex === undefined ||
+      !this.selected.label
+    ) {
+      return;
+    }
+    const offset =
+      this.orientation === 'horizontal'
+        ? this.selected.label.offsetLeft + this.selected.label.offsetWidth
+        : this.selected.label.offsetHeight * (this.selectedIndex + 1) +
+          parseFloat(getComputedStyle(this).getPropertyValue('--sbb-spacing-responsive-m')) *
+            16 *
+            this.selectedIndex;
+    this.style.setProperty('--sbb-stepper-marker-size', `${offset}px`);
+
+    // Remove [data-disable-animation] after the first marker size setting
+    setTimeout(() => this.toggleAttribute('data-disable-animation', false), DEBOUNCE_TIME);
+  }
+
+  private _configure(): void {
+    const labels = this.steps.map((s) => s.label);
+    let posInSet = 0;
+    labels.forEach((label) => {
+      if (label) {
+        label.internals.ariaSelected = (this.selected?.label === label).toString();
+        label.internals.ariaPosInSet = `${++posInSet}`;
+        label.internals.ariaSetSize = `${labels.length}`;
+      }
+    });
+  }
+
+  private _updateLabels(): void {
+    this.steps.forEach((step) => {
+      step.slot = this.orientation === 'horizontal' ? 'step' : 'step-label';
+      step.setAttribute('data-orientation', this.orientation);
+      step.label?.setAttribute('data-orientation', this.orientation);
+    });
+  }
+
+  private _checkOrientation(): void {
+    if (this.horizontalFrom) {
+      this.orientation = isBreakpoint(this.horizontalFrom) ? 'horizontal' : 'vertical';
+      this._updateLabels();
+    }
+    setTimeout(() => this._setMarkerSize(), 0);
+  }
+
+  private _onStepperResize(): void {
+    this._checkOrientation();
+
+    if (this._resizeObserverTimeout) {
+      clearTimeout(this._resizeObserverTimeout);
+    }
```

Comment:
```suggestion
    clearTimeout(this._resizeObserverTimeout);
```
I think `clearTimeout` is null safe